Source/CarthageKit/Project.swift

```diff
@@ -926,6 +913,24 @@ public final class Project {
 					let subdirectoryPath = (CarthageProjectCheckoutsPath as NSString).appendingPathComponent(name)
 					let linkDestinationPath = relativeLinkDestinationForDependencyProject(dependency.project, subdirectory: subdirectoryPath)
 
+					let dependencyCheckoutURLResource = try? dependencyCheckoutURL.resourceValues(forKeys: [
+						.isSymbolicLinkKey,
+						.isDirectoryKey
+					])
+
+					if dependencyCheckoutURLResource?.isSymbolicLink == .some(true) {
+						_ = dependencyCheckoutURL.carthage_path.withCString(Darwin.unlink)
```

Comment:
`Darwin.unlink` differs in not removing directories.

But, since we have the check beforehand, if `removeItem` makes more sense to you?